call-home/src/bin/stats/main.rs

```diff
@@ -0,0 +1,185 @@
+fn initialize_exporter(args: &Cli) {
+    ExporterConfig::initialize(args.metrics_endpoint);
+}
+
+#[tokio::main]
+async fn main() -> errors::Result<()> {
+    let args = Cli::args();
+    utils::print_package_info!();
+    info!(?args, "stats aggregation started");
+
+    init_logging(&args);
+
+    let bus_sub = mbus_init(args.mbus_url.as_str()).await;
+    info!("mbus initialized successfully!");
+
+    let init_data = initialize_events_store(&args.namespace).await?;
+    info!("event store initialized successfully!");
+
+    initialize_events_cache(init_data).await;
+    info!("event cache initialized successfully!");
+
+    initialize_exporter(&args);
+    info!("exporter initialized successfully!");
+
+    // spawn a new task to store the data in cache.
+    tokio::spawn(async move {
+        cache::events_cache::store_events(bus_sub)
+            .await
+            .map_err(|error| {
+                error!(%error, "Error while storing the events to cahce");
+                flush_traces();
+                error
+            })
+    });
+
+    // spawn a new task to update the config map from cache.
+    tokio::spawn(async move {
+        store::events_store::update_config_map_data(&args.namespace, args.update_period.into())
+            .await
+            .map_err(|error| {
+                error!(%error, "Error while persisting the stats to config map from cache");
+                flush_traces();
+                error
+            })
+    });
+
+    let app = move || {
+        actix_web::App::new()
+            .wrap(middleware::Logger::default())
+            .configure(stats_route)
+    };
+
+    HttpServer::new(app)
+        .bind(exporter::exporter_config::ExporterConfig::get_config().metrics_endpoint())
+        .unwrap()
+        .run()
+        .await
+        //.map_err(|e| e.into());
+        .expect("Port should be free to expose the stats");
+
+    Ok(())
+}
+
+/// Initialize logging components -- tracing.
+pub(crate) fn init_logging(args: &Cli) {
+    let tags = default_tracing_tags(raw_version_str(), env!("CARGO_PKG_VERSION"));
+
+    init_tracing("stats", tags, args.jaeger.clone());
+}
+
+fn stats_route(cfg: &mut web::ServiceConfig) {
+    info!(" configuted at /stats");
+    cfg.route("/stats", web::get().to(metrics_handlers));
+}
+
+async fn metrics_handlers() -> impl Responder {
+    // Initialize stats collector
+    let stats_collector = StatsCollector::default();
+    // Create a new registry for prometheus
+    let registry = Registry::default();
+
+    // Register stats collector in the registry
+    if let Err(error) = Registry::register(&registry, Box::new(stats_collector)) {
+        tracing::warn!(%error, "Stats collector already registered");
+    }
+    let mut buffer = Vec::new();
+
+    let encoder = prometheus::TextEncoder::new();
+
+    _ = encoder
+        .encode(&registry.gather(), &mut buffer)
+        .context(errors::CustomMetricsEndodeFailure);
```

Comment:
```suggestion
        .context(errors::CustomMetricsEndodeFailure)?;
```

Reply:
we are not returning any error here . so `?` not used . Similar implementation is there in ecporter . https://github.com/openebs/mayastor-extensions/blob/3d8cb712273fa2ca3d80fea22f2854aecadbc95a/exporter/src/pool/bin/main.rs#L134

Reply:
What do you mean we are not returning an error here?
We are, except we're always returning Ok: `HttpResponse::Ok()`, even when we fail to encode.